demos/gesture_recognition_demo/cpp_gapi/include/stream_source.hpp

```diff
@@ -0,0 +1,120 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <utils/images_capture.h>
+#include <opencv2/gapi.hpp>
+#include <chrono>
+
+namespace custom {
+class CustomCapSource : public cv::gapi::wip::IStreamSource
+{
+public:
+    explicit CustomCapSource(std::shared_ptr<ImagesCapture>& cap, const int batch_size, const int batch_fps)
+    : cap(cap), batch_size(batch_size), batch_fps(batch_fps) {
+        prep();
+    }
+
+protected:
+    std::shared_ptr<ImagesCapture> cap;
+    int batch_size = 0; // size of batch
+    int batch_fps = 0;  // constant FPS for batch
+    cv::Mat batch;
```

Comment:
I believe this `Mat` should be commented too, especially if this `Mat`  is not a plain simple `Mat` but hides some "batch" inside. What is its memory layout then?